scripts/eventDataScript.js

```diff
@@ -192,98 +194,74 @@ function deleteSpecificTeam()
 
     var season = getCookie('currCheckedSeason');
     var currEvent = getCurrEvent(); 
+    var teamsJSON = getTeams(season,currEvent);
     var cname = event_data_cookie_name(season,currEvent);
-    var teamsJSON = JSON.parse(getCookie(cname));
 
     delete teamsJSON[input];
     setCookie(cname,JSON.stringify(teamsJSON),750);
     setTeamTable(season,currEvent);
     $("#deleteSpecificTeamTextBox").val("").focus();
 }
 
+function saveTeam() //onclick for saving the teams
+{
+    var currSeason = getCookie('currCheckedSeason');
+    var currEvent = getCurrEvent();
+    var teamNumber = parseInt($("#teamNumber").val().trim());
+    var teamName = $("#teamName").val().trim();
+    var teams = {};
+
+    if (teamNumber === "" || teamName === "")
```

Comment:
`teamNumber` will never triple equal empty string, triple equals checks type as well.

if #teamNumber.val() is empty,
`var teamNumber = parseInt($("#teamNumber").val().trim());`
will yield `NaN` (not a number), to check for this, use `Number.isNaN()`:
```js
if(Number.isNaN(teamNumber) || teamName === "")
```